service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderController.java

```diff
@@ -89,68 +81,45 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody Call
     }
 
     @PostMapping("/mid-event")
-    public AboutToStartOrSubmitCallbackResponse handleMidEvent(
-        @RequestBody CallbackRequest callbackRequest) {
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackRequest) {
+        Map<String, Object> data = callbackRequest.getCaseDetails().getData();
+        CaseData caseData = mapper.convertValue(data, CaseData.class);
+        Document document = caseManagementOrderService.getDocument(caseData);
 
-        CaseDetails caseDetails = callbackRequest.getCaseDetails();
-        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
-
-        Document document = getDocument(caseData, true);
-
-        caseDetails.getData()
-            .put(ORDER_ACTION.getKey(), OrderAction.builder().document(buildFromDocument(document)).build());
+        data.put(ORDER_ACTION.getKey(), OrderAction.builder().document(buildFromDocument(document)).build());
 
         return AboutToStartOrSubmitCallbackResponse.builder()
-            .data(caseDetails.getData())
+            .data(data)
             .build();
     }
 
-    //TODO: refactor. far too much logic in this controller now FPLA-1469
     @PostMapping("/about-to-submit")
-    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(
-        @RequestBody CallbackRequest callbackRequest) {
-
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackRequest) {
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
         CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+        CaseManagementOrder order = caseData.getCaseManagementOrder();
 
-        if (caseData.getCaseManagementOrder() == null || !caseData.getCaseManagementOrder().isInJudgeReview()) {
```

Comment:
Can't the CMO still be null on the first pass through

Reply:
Take it back, thought I was in the draft cmo controller for some reason

Reply:
Actually revert that take back, the judge and LA have access to their events in the same state, so judge may for some reason be trying to access the event before the LA has done their thing, as such the object could still be null I think

Reply:
so, there is the show hide page condition in ccd which should block your above. I am fairly certain cmo will always exist at this point (as it is made in draft controller right?)

Reply:
But the about to submit callback will still be called right? They still enter the event to see that page.

Reply:
Unless they cannot progress past that page, haven't checked the flow in some time